tests/unittests/InterpreterTest.cpp

```diff
@@ -898,6 +898,90 @@ TEST(Interpreter, nonLinearClassifier) {
   }
 }
 
+/// Generate images in two classes. A "line" is labeled as 0
+/// and a "cross" is labeled as 1
+static void generateImageData(Tensor &images, Tensor &labels, unsigned numSamples, unsigned isTrain) {
+  auto L = labels.getHandle<size_t>();
+  auto image = images.getHandle<>();
+  images.zero();
+  
+  for (size_t i = 0; i < numSamples; i++) {
+    bool isLine = i % 2 == 0;
+    L.at({i, 0}) = isLine ? 0 : 1;
+    size_t target;
+    // to make sure testing set is not totally included in 
+    // training set
+    if (isTrain)  {
+      target = rand() % 25 + 1;
+    }
+    else {
+      target = rand() % 30 + 1;
+    }
+    for (size_t x = 0; x < 32; x++) {
```

Comment:
since target value is known you could just do single loop here to fill in the tensor.

Reply:
@beicy i think this is not yet addressed.